frontend/src/concepts/pipelines/elyra/utils.ts

```diff
@@ -105,3 +118,49 @@ export const generateElyraServiceAccountRoleBinding = (
     },
   ],
 });
+
+export const createElyraServiceAccountRoleBinding = async (
+  notebook: NotebookKind,
+): Promise<RoleBindingKind | void> => {
+  const notebookName = notebook.metadata.name;
+  const namespace = notebook.metadata.namespace;
+  const notebookUid = notebook.metadata.uid;
+
+  // Check if rolebinding is already exists for backward compatibility
+  const roleBinding = await getRoleBinding(
+    namespace,
+    getElyraServiceAccountRoleBindingName(notebookName),
+  ).catch((e) => {
+    // 404 is not an error
+    if (e.statusObject?.code === 404) {
+      return undefined;
+    }
+    throw e;
```

Comment:
You don't thrown an error when the role binding fails to create but you throw an error failing to fetch? We should do one or the other whether it's required to have a role binding or not.
In this case it seems to me like any error should continue down the creation path.

Reply:
The only reason to catch here is to make sure whether the role binding is already there, so we can make it backward-compatible. The `throw` here is to avoid the `Not all code paths return a value` type error.

Reply:
@DaoDaoNoCode But why throw instead of continuing the flow and attempting to create the role binding?

Reply:
Ah, yeah, I might think it too complex, updated, thanks!